pkg/controllers/subnetport/subnetport_controller.go

```diff
@@ -118,7 +119,27 @@ func (r *SubnetPortReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 			r.StatusUpdater.UpdateFail(ctx, subnetPort, err, fmt.Sprintf("Failed to get Subnet by path: %s", nsxSubnetPath), setSubnetPortReadyStatusFalse, r.SubnetPortService)
 			return common.ResultRequeue, err
 		}
-		nsxSubnetPortState, err := r.SubnetPortService.CreateOrUpdateSubnetPort(subnetPort, nsxSubnet, "", labels)
+
+		isVmSubnetPort := true
+		if subnetPort.Labels[servicecommon.LabelImageFetcher] == "true" {
```

Comment:
change to if fetcherValue, ok := subnetPort.Labels[servicecommon.LabelImageFetcher]; ok { ... }

Reply:
ok